spark/src/main/java/org/apache/iceberg/spark/source/IcebergSource.java

```diff
@@ -104,7 +104,7 @@ public DataSourceReader createReader(StructType readSchema, DataSourceOptions op
     validateWriteSchema(table.schema(), dsSchema, checkNullability(options));
     validatePartitionTransforms(table.spec());
     String appId = lazySparkSession().sparkContext().applicationId();
-    String wapId = lazySparkSession().conf().get("spark.wap.id", null);
+    String wapId = options.get("wap-id").orElse(lazySparkSession().conf().get("spark.wap.id", null));
```

Comment:
Can we move the WAP changes to a separate PR?

Reply:
+1